From 0fe9042e22b8c8c05d0d3aed550d18b511e0abd5 Mon Sep 17 00:00:00 2001
From: josselinonduty <contact@josselinonduty.fr>
Date: Fri, 31 Jan 2025 17:56:13 +0100
Subject: [PATCH] feat: add cli argument to disable discord rpc

---
 build/main.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/main.js b/build/main.js
index 69ce1f1..8b645c6 100644
--- a/build/main.js
+++ b/build/main.js
@@ -1235,6 +1235,7 @@
         this.mprisPlayer.on('raise', () => this.app.getWindow().show())
       }
       initDiscordRichPresence() {
+        if (process.argv.some(arg => arg === '--disable-discord-rpc')) return;
         rpcConnection = new external_rich_presence_builder_namespaceObject({
           clientID: "1244016234203185183",
         });
-- 
2.43.0

