diff --git a/build/main.js b/build/main.js
index 5066eef..46b04c5 100644
--- a/build/main.js
+++ b/build/main.js
@@ -3083,7 +3083,10 @@
             this._adjustViewToContentSize(),
             mainView.setAutoResize({ width: !0, height: !0 });
         } else this.appService.setWindow(this.window, void 0);
-        if ((this.appService.setUserAgent(), isPlatform(PLATFORM.DARWIN))) {
+        if (
+             (this.appService.setUserAgent(), isPlatform(PLATFORM.LINUX))
+             && !process.argv.some(arg => arg === '--disable-systray')
+        ) {
           let isQuitting = !1;
           external_electron_namespaceObject.app.on(
             "before-quit",
